lld/MachO/MapFile.cpp

```diff
@@ -220,7 +234,12 @@ void macho::writeMapFile() {
   os << "# Address\tSize    \tFile  Name\n";
   for (const OutputSegment *seg : outputSegments) {
     for (const OutputSection *osec : seg->getSections()) {
-      if (auto *concatOsec = dyn_cast<ConcatOutputSection>(osec)) {
+      const TextOutputSection *textOsec = dyn_cast<TextOutputSection>(osec);
+      if (textOsec && textOsec->getThunks().size()) {
```

Comment:
I think this will still work even if `getThunks()` is empty
```suggestion
      if (auto *textOsec = dyn_cast<TextOutputSection>(osec)) {
```